scripts/mbedtls_framework/build_tree.py

```diff
@@ -122,3 +129,13 @@ def guess_tf_psa_crypto_root(root: Optional[str] = None) -> str:
         return root
     else:
         raise Exception('TF-PSA-Crypto source tree not found')
+
+def is_mbedtls_3_6_branch(root: Optional[str] = None) -> bool:
+    """Whether root is the root directory of the Mbed TLS 3.6 branch """
+    if root is None:
+        root = os.getcwd()
```

Comment:
why set it to none and check it in the body of the method while we could be setting the `os.getcwd()` as the default?

Also I would mention in the docstring that the method will try to check the current directory. 

I am personally not a fan of using getcdw() since that stops method from working when invoked by other directories and other scripts.

Reply:
I've removed the `root` parameter that I found eventually just confusing.